torch/_inductor/graph.py

```diff
@@ -920,6 +921,54 @@ def set_current_node(self, node: torch.fx.Node):
         finally:
             self.current_node = old
 
+    def match_insignificant_strides(
+        self,
+        tensor: ir.TensorBox,
+        meta_strides_inp: Tuple[Union[int, torch.SymInt], ...],
+    ) -> ir.TensorBox:
+        # should have already been realized
+        assert torch._inductor.ir.is_storage_and_layout(tensor)
+
+        meta_strides = [
+            s.node.expr if isinstance(s, torch.SymInt) else s for s in meta_strides_inp
+        ]
+
+        if all(
+            self.sizevars.statically_known_equals(s1, s2)
+            for s1, s2 in zip(meta_strides, tensor.get_stride())
+        ):
+            return tensor
+
+        def significant_strides_equal(shape, meta_strides, tensor_strides):
+            for dim, s1, s2 in zip(shape, meta_strides, tensor_strides):
+                if self.sizevars.statically_known_leq(dim, 1):  # type: ignore[arg-type]
+                    continue
+
+                if not self.sizevars.statically_known_equals(s1, s2):
+                    return False
+
+            return True
+
+        if not significant_strides_equal(
```

Comment:
Get confused. Why we return directly here if significant strides are not equal?

Reply:
We might have actual layout difference bc of things like channels last. In that case the `"did the strides diverge between inductor and eager"` is going to fire regardless of the size0/1 dimension stride checks

Reply:
I'll add comment